src/app/nonprofit/nonprofit-page/donation-dialog.html

```diff
@@ -0,0 +1,124 @@
+<button mat-icon-button class="close-icon-button" [mat-dialog-close]="true">
+    <mat-icon class="close-icon">close</mat-icon>
+</button>
+<h1 mat-dialog-title>
+    Thank you for choosing to to donate to {{data.nonprofitName}}!
+</h1>
+<div mat-dialog-content class="donation-stepper-container">
+    <mat-vertical-stepper linear #stepper>
+        
+        <!-- Donation Amount Stepper-->
+        <mat-step [stepControl]="donationAmountForm">
+            <form [formGroup]="donationAmountForm">
+                <ng-template matStepLabel>Enter Donation Amount</ng-template>
+
+                <div class="donation-inputs-container">
+                    <!-- Input for donation amount -->
+                    <div class="donation-input">
+                        <mat-form-field>
+                            <mat-label>Enter Amount to Donate</mat-label>
+                            <input formControlName="donationAmount" matInput type="tel" currencyMask>
+                        </mat-form-field>
+                    </div>
+                </div>
+                <button mat-button matStepperNext>Next</button>
+            </form>
+        </mat-step>
+
+        <!-- Your Information Stepper -->
+        <mat-step [stepControl]="donorInformationForm">
+            <form [formGroup]="donorInformationForm">
+                <ng-template matStepLabel>Your Information</ng-template>
+                
+                <div class="user-information-parent">
+                    <div class="inputs-container">
+                        <div class="donation-input">
+                            <!-- Card Holder Name -->
+                            <mat-form-field>
+                                <mat-label>Card Holder Name</mat-label>
+                                <input formControlName="cardHolderName" matInput type="text" >
+                            </mat-form-field> 
+                            
+                            <!-- Expiration Date-->
+                            <mat-form-field>
+                                <mat-label>Card Expiration Date</mat-label>
+                                <input formControlName="cardExpirationDate" matInput type="tel" ccExp>
+                            </mat-form-field>
+
+                        </div>
+                    </div>
+                    <div class="inputs-container">
+                        <div class="donation-input">
+                            <!-- Card Number -->
+                            <mat-form-field>
+                                <mat-label>Card Number</mat-label>
+                                <input formControlName="cardNumber" matInput type="tel" ccNumber>
```

Comment:
So this is actually expected. CC numbers have special initial identifiers. The library uses some mapping of the first digit of the CC number and creates a mask that is appropriate to that CC. For example, if you start a CC number with the number 4, then it will create a mask for VISA CC's. However, MasterCard CC numbers start with 5. If MasterCard and Visa have different length CC's, then the library will adjust the masking to fit the requirements of that CC. But you input a random number like 1 (no CC starts with 1) then it will NOT create a mask because 1 does not map to any specific CC masking pattern in the library. Hope that made sense!

Reply:
Ahhh great that makes a lot of sense, thanks